src/components/ImageView/index.native.js

```diff
@@ -198,12 +201,15 @@ class ImageView extends PureComponent {
                                     scale: 2,
                                     duration: 100,
                                 });
+
+                                this.props.onScaleChanged(2);
```

Comment:
`onMove` will be called after the zoom animation, so it's possible to zoom and swipe and stuck in between the images. Sending scale just when we actually trigger the animation makes this nearly impossible, you should be really fast to catch in between state updates. And this lucky case will be fixed by migration to UI thread only code with gesture handler and reanimated.